tensorrt_llm/_torch/auto_deploy/transformations/_graph.py

```diff
@@ -106,11 +122,30 @@ def move_to_device(gm: fx.GraphModule, device: DeviceLikeType) -> fx.GraphModule
             kwargs = node.kwargs.copy()
             kwargs["device"] = device
             node.kwargs = kwargs
+
+        if is_op(node, torch.ops.aten.to.device):
+            args = list(node.args)
+            args[1] = device
+            node.args = tuple(args)
+
         # move all the tensor metadata
         node.meta["val"] = pytree.tree_map(
             lambda v: v.to(device) if isinstance(v, torch.Tensor) else v,
             node.meta.get("val"),
         )
+    if recompile_graph:
+        # recompile graph to update self generated codes in subgraph
+        gm.graph.lint()
+        gm.recompile()
+
+
+def move_to_device(gm: fx.GraphModule, device: DeviceLikeType) -> fx.GraphModule:
+    """Move the entire graph module and all sub-GraphModules to the specified device."""
+    # get device
+    device = torch.device(device)
+
+    for _, subgm in named_graphmodules(gm):
+        _move_single_gm_to_device(subgm, device, subgm is not gm)
```

Comment:
nit: please add a comment to why the recompile_graph is necessary just as you explained in the other comment